driver/controller.go

```diff
@@ -81,6 +87,21 @@ func (s controllerService) CreateVolume(ctx context.Context, req *csi.CreateVolu
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	// check if the create volume request has a data source
+	if source != nil {
+		// get the parent volumeID/snapshotID if exists
+		parentVol, parentID, err = s.validateContentSource(ctx, req)
+		if err != nil {
+			return nil, err
+		}
+		// check if the child volume has the same size as the parent volume.
+		// TODO (Yuggupta27): Allow user to create a child volume with more size than that of the parent.
+		parentSizeGb := parentVol.Spec.Size.Value() >> 30
+		if parentSizeGb != requestGb {
+			return nil, status.Error(codes.InvalidArgument, "requested size does not match parent size")
```

Comment:
Does this error code need to be changed? From the [CSI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md#createvolume):

> If CO requests a volume to be created from existing snapshot or volume and the requested size of the volume is larger than the original snapshotted (or cloned volume), the Plugin can either refuse such a call with OUT_OF_RANGE error or MUST provide a volume that, when presented to a workload by NodePublish call, has both the requested (larger) size and contains data from the snapshot (or original volume). Explicitly, it's the responsibility of the Plugin to resize the filesystem of the newly created volume at (or before) the NodePublish call, if the volume has VolumeCapability access type MountVolume and the filesystem resize is required in order to provision the requested capacity.

Reply:
Thanks! updated :+1: